opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts

```diff
@@ -135,3 +146,49 @@ export const workspaceEditContext = async (context: AuthContext, user: AuthUser,
   return storeLoadById(context, user, workspaceId, ENTITY_TYPE_WORKSPACE)
     .then((workspaceToReturn) => notify(BUS_TOPICS[ENTITY_TYPE_WORKSPACE].EDIT_TOPIC, workspaceToReturn, user));
 };
+
+export const checkDashboardConfigurationImport = (parsedData: any) => {
+  if (parsedData.type !== 'dashboard') {
+    throw FunctionalError('Invalid type. Please import OpenCTI dashboard-type only', { reason: parsedData.type });
+  }
+
+  const MINIMAL_COMPATIBLE_VERSION = '5.12.0';
+  const isCompatibleOpenCtiVersion = (openCtiVersion: string) => {
+    const [major, minor, patch] = openCtiVersion.split('.').map((number) => parseInt(number, 10));
+    const [openCtiMajor, openCtiMinor, openCtiPatch] = MINIMAL_COMPATIBLE_VERSION.split('.').map((number) => parseInt(number, 10));
+    return major >= openCtiMajor && minor >= openCtiMinor && patch >= openCtiPatch;
+  };
+
+  if (!isCompatibleOpenCtiVersion(parsedData.openCTI_version)) {
+    throw FunctionalError(`Invalid version of the platform. Please upgrade your OpenCTI. Minimal version required: ${MINIMAL_COMPATIBLE_VERSION}`, { reason: parsedData.openCTI_version });
+  }
+};
+
+export const workspaceImport = async (context: AuthContext, user: AuthUser, file: Promise<FileHandle>) => {
+  const authorizedMembers = initializeAuthorizedMembers([], user);
+  const uploadedFile = await file;
+  const readStream = uploadedFile.createReadStream();
+  const fileContent = await streamToString(readStream);
+  const parsedData = JSON.parse(fileContent.toString());
+
+  const mappedData = {
+    type: parsedData.type,
+    openCTI_version: parsedData.openCTI_version,
+    name: parsedData.configuration.name,
+    manifest: parsedData.configuration.manifest,
+    authorized_members: authorizedMembers,
+  };
+  checkDashboardConfigurationImport(mappedData);
+  const importWorkspaceCreation = await createEntity(context, user, mappedData, ENTITY_TYPE_WORKSPACE);
+  const workspaceId = importWorkspaceCreation.id;
+  await publishUserAction({
+    user,
+    event_type: 'mutation',
+    event_scope: 'create',
+    event_access: 'extended',
+    message: `import ${importWorkspaceCreation.name} workspace`,
+    context_data: { id: workspaceId, entity_type: ENTITY_TYPE_WORKSPACE, input: importWorkspaceCreation }
+  });
+  await notify(BUS_TOPICS[ENTITY_TYPE_WORKSPACE].ADDED_TOPIC, importWorkspaceCreation, user);
+  return workspaceId;
```

Comment:
also, to be more consistent with `workspaceAdd` mutation, shouldn't we return the workspace object, not only workspaceId ?

Reply:
No it was the point to return just the id

Reply:
We have chosen this approach based on our need and spotted upload in the codebase